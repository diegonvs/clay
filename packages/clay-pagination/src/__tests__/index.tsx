/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import * as React from 'react';
import ClayPagination from '..';
import {
	cleanup,
	fireEvent,
	getByTestId,
	getByText,
	render,
} from 'react-testing-library';

const spritemap = 'path/to/spritemap';

describe('ClayPagination', () => {
	afterEach(cleanup);

	it('renders', () => {
		const {container} = render(
			<ClayPagination
				activePage={12}
				spritemap={spritemap}
				totalPages={25}
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it('calls onPageChange when arrow is clicked', () => {
		const changeMock = jest.fn();

		const {container} = render(
			<ClayPagination
				activePage={12}
				onPageChange={changeMock}
				spritemap={spritemap}
				totalPages={25}
			/>
		);

		fireEvent.click(
			getByTestId(container, 'prevArrow') as HTMLButtonElement,
			{}
		);

		expect(changeMock).toHaveBeenLastCalledWith(11);

		fireEvent.click(
			getByTestId(container, 'nextArrow') as HTMLButtonElement,
			{}
		);

		expect(changeMock).toHaveBeenLastCalledWith(13);
	});

	it('calls onPageChange when individual page is clicked', () => {
		const changeMock = jest.fn();

		const {container} = render(
			<ClayPagination
				activePage={12}
				onPageChange={changeMock}
				spritemap={spritemap}
				totalPages={25}
			/>
		);

		fireEvent.click(getByText(container, '25') as HTMLElement, {});

		expect(changeMock).toHaveBeenLastCalledWith(25);
	});

	it('shows dropdown when ellipsis is clicked', () => {
		const {container} = render(
			<ClayPagination
				activePage={12}
				spritemap={spritemap}
				totalPages={25}
			/>
		);

		fireEvent.click(getByText(container, '...') as HTMLElement, {});

		expect(
			document.body.querySelector('.dropdown-menu')!.classList
		).toContain('show');
	});

	it('calls onPageChange when iitem is clicked in dropdown-menu', () => {
		const changeMock = jest.fn();

		const {container} = render(
			<ClayPagination
				activePage={12}
				onPageChange={changeMock}
				spritemap={spritemap}
				totalPages={25}
			/>
		);

		fireEvent.click(getByText(container, '...') as HTMLElement, {});

		fireEvent.click(getByText(document.body, '4') as HTMLElement, {});

		expect(changeMock).toHaveBeenLastCalledWith(4);
	});
});
