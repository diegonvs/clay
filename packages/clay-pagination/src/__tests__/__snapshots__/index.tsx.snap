// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayPagination renders 1`] = `
<div>
  <ul
    class="pagination pagination-root"
  >
    <li
      class="page-item"
    >
      <button
        class="page-link"
        data-testid="prevArrow"
      >
        <svg
          class="lexicon-icon lexicon-icon-angle-left"
          role="presentation"
        >
          <use
            xlink:href="path/to/spritemap#angle-left"
          />
        </svg>
      </button>
    </li>
    <li
      class="page-item"
    >
      <button
        class="page-link"
      >
        1
      </button>
    </li>
    <li
      class="dropdown page-item"
    >
      <span
        style="display: inline-block;"
      >
        <button
          class="dropdown-toggle page-link btn btn-unstyled"
          type="button"
        >
          ...
        </button>
      </span>
    </li>
    <li
      class="page-item"
    >
      <button
        class="page-link"
      >
        10
      </button>
    </li>
    <li
      class="page-item"
    >
      <button
        class="page-link"
      >
        11
      </button>
    </li>
    <li
      class="page-item active"
    >
      <button
        class="page-link"
      >
        12
      </button>
    </li>
    <li
      class="page-item"
    >
      <button
        class="page-link"
      >
        13
      </button>
    </li>
    <li
      class="page-item"
    >
      <button
        class="page-link"
      >
        14
      </button>
    </li>
    <li
      class="dropdown page-item"
    >
      <span
        style="display: inline-block;"
      >
        <button
          class="dropdown-toggle page-link btn btn-unstyled"
          type="button"
        >
          ...
        </button>
      </span>
    </li>
    <li
      class="page-item"
    >
      <button
        class="page-link"
      >
        25
      </button>
    </li>
    <li
      class="page-item"
    >
      <button
        class="page-link"
        data-testid="nextArrow"
      >
        <svg
          class="lexicon-icon lexicon-icon-angle-right"
          role="presentation"
        >
          <use
            xlink:href="path/to/spritemap#angle-right"
          />
        </svg>
      </button>
    </li>
  </ul>
</div>
`;
